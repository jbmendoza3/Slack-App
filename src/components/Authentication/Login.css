* {
  padding: 0;
  margin: 0;

  font-family: "Roboto", sans-serif;
  font-weight: 400;
  font-style: normal;

  transition-duration: 0.3s;
}

h1 {
  font-weight: 700;
  font-size: 2.7rem;
}

label {
  font-size: 1.2rem;
  width: 20rem;
}

button{
  cursor: pointer;
}

input[type="text"],
input[type="password"],
input[type="email"] {
  border: hidden;
  background-color: rgb(237, 237, 237);
  font-size: 1.2rem;
  margin-bottom: 1rem;
  padding: 0.5rem;
  border-radius: 0.5rem;
  width: 20rem;
}

input[type="text"]:focus,
input[type="password"]:focus,
input[type="email"]:focus  {
  outline: none;
  background-color: rgb(238, 223, 255);
}


.main-container {
  display: flex;
  flex-wrap: wrap;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}

.header2 {
  margin-block: 1rem;
  font-size: 1.2rem;
}

.logo-img {
  margin: 2rem;
  margin-top: 2.7rem;
  max-width: 100px;
}

.login-form {
  padding: 1.3rem;
  background-color: rgb(255, 255, 255);
  width: 30rem;
  border-radius: 1rem;
  box-shadow: 0 0 10px 0 rgba(0, 0, 0, 0.5);
}

.login-btn {
  padding-inline: 2rem;
  padding-block: 0.6rem;
  border-radius: 0.7rem;
  border: hidden;
  background-color: rgb(29, 29, 29);
  font-size: large;
  color: white
}

.login-btn:hover {
    cursor: pointer;
    background-color: rgb(50, 50, 50);
    transition-duration: 0.2s;
}

.changeDisplayLogin-btn {
  border: hidden;
  margin-top: 1rem;
  background-color: transparent;
  color: rgb(132, 17, 225);
  text-decoration: underline;
}
