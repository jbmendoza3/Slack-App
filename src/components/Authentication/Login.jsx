import React, { useState } from "react";
import "./Login.css";
import logo from "../../assets/slack-logo.png";
import { useNavigate } from "react-router-dom";
import axios from 'axios';
import { API_URL } from "../../constants/Constants";

export default function Login({ setDisplay }) {
   const [email, setEmail] = useState();
   const [password, setPassword] = useState();


   const Navigate = useNavigate();
   const handleLogin = async (event) => {
      event.preventDefault();
      if (!email || !password) {
         return alert("Please input your Login Credentials");
      }

      try {
         const loginCredentials = {
            email,
            password,
         };

         const response = await axios.post(
            `${API_URL}/auth/sign_in`,
            loginCredentials
         );
         const { data, headers } = response;

         if (data && headers) {
            const accessToken = headers["access-token"];
            const client = headers.client;
            const expiry = headers.expiry;
            const uid = headers.uid;

            const necessaryData = {
               accessToken,
               client,
               expiry,
               uid,
               id: data.data.id,
            };

            localStorage.setItem("user", JSON.stringify(necessaryData));
            localStorage.setItem("isLoggedIn", "true");
            Navigate("/dashboard");
         }
      } catch (error) {
         if (error.response.data.errors) {
            return alert(error.response.data.errors);
         }
      }
   };

   return (
      <div className="main-container">
         <img src={logo} alt="Slack" className="logo-img" />
         <div className="login-form">
            <h1>Sign in to Slack</h1>
            <p className="header2">
               We suggest using the email address you use at work.
            </p>
            <form onSubmit={handleLogin}>
               <label>Email</label>
               <br></br>
               <input
                  type="email"
                  onChange={(event) => setEmail(event.target.value)}
                  required
               />
               <br></br>
               <label>Password</label>
               <br></br>
               <input
                  type="password"
                  onChange={(event) => setPassword(event.target.value)}
                  required
               />
               <br></br>
               <input type="submit" value="Login" className="login-btn" />
            </form>
            <button
               className="changeDisplayLogin-btn"
               onClick={() => setDisplay("signup")}
            >
               Create an Account
            </button>
         </div>
      </div>
   );
}
