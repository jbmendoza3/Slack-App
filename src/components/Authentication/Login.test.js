import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import Login from './Login';
import renderer from "react-test-renderer";

test('renders without crashing', () => {
  render(
    <Router>
      <Login />
    </Router>
  );
});

test("snapshot for login", () => {
  const login = renderer.create(
    <Router>
      <Login />
    </Router>
  ).toJSON();
  expect(login).toMatchSnapshot();
});
