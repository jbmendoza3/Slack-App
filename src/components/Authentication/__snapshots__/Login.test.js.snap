// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot for login 1`] = `
<div
  className="main-container"
>
  <img
    alt="Slack"
    className="logo-img"
    src="slack-logo.png"
  />
  <div
    className="login-form"
  >
    <h1>
      Sign in to Slack
    </h1>
    <p
      className="header2"
    >
      We suggest using the email address you use at work.
    </p>
    <form
      onSubmit={[Function]}
    >
      <label>
        Email
      </label>
      <br />
      <input
        onChange={[Function]}
        required={true}
        type="email"
      />
      <br />
      <label>
        Password
      </label>
      <br />
      <input
        onChange={[Function]}
        required={true}
        type="password"
      />
      <br />
      <input
        className="login-btn"
        type="submit"
        value="Login"
      />
    </form>
    <button
      className="changeDisplayLogin-btn"
      onClick={[Function]}
    >
      Create an Account
    </button>
  </div>
</div>
`;
